### ISSUE TITLE
[Bug]: sometimes getting unresolved external symbols when generating source

### ISSUE THREAD

**Issue by EnokViking** [beefytech/Beef#1812]
I have a class `VertexArray<T> where T : struct` and internally I resort to source generation to generate vertex attribute layouts based on the field ordering and type of the input T and every time I change T to a different type and try to compile, the compilation sometimes fails with:
```
vdata.obj : error LNK2001: unresolved external symbol "protected: class bf::System::Object * __cdecl bf::StonecutterFramework::App::App::DynamicCastToTypeId(int)" (?DynamicCastToTypeId@App@App@StonecutterFramework@bf@@IEAAPEAVObject@System@4@H@Z)
vdata.obj : error LNK2001: unresolved external symbol "protected: class bf::System::Object * __cdecl bf::StonecutterFramework::App::App::DynamicCastToInterface(int)" (?DynamicCastToInterface@App@App@StonecutterFramework@bf@@IEAAPEAVObject@System@4@H@Z)
```
and sometimes it complains about syntax errors in the emitted source, even though inspecting it through the 'Emit'-window reveals the source is just fine.

In SFTestbed/EntryPoint.bf, just uncomment / comment the vertex arrays and do the same for the VertexArray declarations and hit compile.

I can circumvent this by hitting Debug Comptime or cleaning prior to building / debugging after every time I change T.
There's a comment about another unrelated(i think?) bug at line 108 in VertexArray.bf (i can make a separate issue for this if you want)
[BuggyComptime.zip](https://github.com/beefytech/Beef/files/10994012/BuggyComptime.zip)

**Comment by bfiete** [comment]
I couldn't figure out what the bug was supposed to be at line 108 in VertexArray.bf.

What is the code that used `FieldInfo.FieldType` "down there" that didn't work?

**Comment by EnokViking** [comment]
Actually, that was nothing I just derped. I tried to do `t.GetField(i).FieldType` but GetField returns a Result<T>, sorry for the confusion.

**Comment by bfiete** [comment]
Linking error fixed at 7ab6800f40b735875035b4cc901e609f6420a833
